components/sync15/src/request.rs

```diff
@@ -1242,11 +1257,114 @@ mod test {
             request_bytes_for_payloads(&[100])
         );
     }
+ 
+    #[test]
+    fn test_pq_multi_batch_records_and_bytes() {
```

Comment:
I'm not really clear what's being tested here. Can you walk me through what you're validating here?

Reply:
The two tests above it test record limits for batches and byte limits for batches. This tests essentially combines them and tests record and byte limit cases together--both combinations that should succeed and combinations that should not. This test was requested in the TODO section at the bottom.